cmd/README.md

```diff
@@ -0,0 +1,639 @@
+# Precompile Generation Tutorial
+
+We can now generate a stateful precompile with the Precompile gen tool!
+
+### Assumption of Knowledge
+Before starting this tutorial it would be helpful if you had some context on the EVM, precompiles, and stateful precompiles. 
+Here are some resources to get started put together. 
+
+- [The Ethereum Virtual Machine](https://github.com/ethereumbook/ethereumbook/blob/develop/13evm.asciidoc)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Deconstructing a Smart Contract](https://blog.openzeppelin.com/deconstructing-a-solidity-contract-part-i-introduction-832efd2d7737/)
+- [Layout of State Variables in Storage](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_storage.html)
+- [Layout in Memory](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_memory.html)
+- [Layout of Call Data](https://docs.soliditylang.org/en/v0.8.10/internals/layout_in_calldata.html)
+- [Contract ABI Specification](https://docs.soliditylang.org/en/v0.8.10/abi-spec.html)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+- [Precompiles in Solidity](https://medium.com/@rbkhmrcr/precompiles-solidity-e5d29bd428c4)
+- [Customizing the EVM with Stateful Precompiles](https://medium.com/avalancheavax/customizing-the-evm-with-stateful-precompiles-f44a34f39efd)
+
+ 
+## Tutorial
+
+Let's start by creating the Solidity interface that we want to implement. We can put this  in `./contract-examples/contracts`
+
+```
+// (c) 2022-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.8.0;
+
+interface IHelloWorld {
+ // sayHello returns the string located at [key]
+  function sayHello() external returns (string calldata);
+
+// setGreeting sets the string located at [key]
+  function setGreeting(string calldata response) external;
+}
+```
+
+Now we have an interface that our precompile can implement!
+Let's create an [abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html#:~:text=Contract%20ABI%20Specification-,Basic%20Design,as%20described%20in%20this%20specification.) of our solidity code.
+
+In the same `./contract-examples/contracts` directory, let's download solc and run
+```
+brew update
+brew tap ethereum/ethereum
+brew install solidity
+
+solc --abi IHelloWorld.sol -o .
+```
+
+This spits out the abi code. Let's move it into a brand new folder in 
+`./contract-examples/contracts/contract-abis`. 
```

Comment:
Can we ask users to keep the ABI file in the same folder instead of creating a new one ie. `./contract-examples/contracts/` ?